cartoframes/data/clients/bigquery_client.py

```diff
@@ -52,6 +60,37 @@ def upload_dataframe(self, dataframe, schema, tablename, project, dataset):
 
     @refresh_client
     def query(self, query, **kwargs):
-        response = self.client.query(query, **kwargs)
+        return self.client.query(query, **kwargs)
+
+    def download_to_file(self, project, dataset, table, limit=None, offset=None, file_path=None, fail_if_exists=False):
+        if not file_path:
+            file_name = '{}.{}.{}.csv'.format(project, dataset, table)
+            file_path = os.path.join(_USER_CONFIG_DIR, file_name)
+
+        if fail_if_exists and os.path.isfile(file_path):
+            raise CartoException('The file `{}` already exists.'.format(file_path))
+
+        query = _download_query(project, dataset, table, limit, offset)
+        rows_iter = self.query(query).result()
+
+        progress_bar = tqdm.tqdm_notebook(total=rows_iter.total_rows)
+
+        with open(file_path, 'w') as csvfile:
+            csvwriter = csv.writer(csvfile)
+            for row in rows_iter:
+                csvwriter.writerow(row.values())
+                progress_bar.update(1)
+
+        warn('Data saved: {}'.format(file_path))
+
+
+def _download_query(project, dataset, table, limit=None, offset=None):
+    full_table_name = '`{}.{}.{}`'.format(project, dataset, table)
+    query = 'SELECT * FROM {}'.format(full_table_name)
```

Comment:
We will use this for both downloading a plain dataset or a geographies dataset. I'd like to be sure of the geometry format downloaded and written in the CSV.

Could we add a test to make sure that we can read/upload to CARTO the geometry in the downloaded CSV file for a geography table? (or any other type of test that you think it might be interesting for this case)